bpf-prog/ipv6-hp-bpf/pkg/ingress/ingress.go

```diff
@@ -0,0 +1,57 @@
+package ingress
+
+import (
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+	"go.uber.org/zap"
+)
+
+// SetupIngressFilter sets up the ingress filter
+func SetupIngressFilter(ifaceIndex int, objs *IngressObjects, logger *zap.Logger) error {
+	link, err := netlink.LinkByIndex(ifaceIndex)
+	if err != nil {
+		logger.Error("Failed to get link", zap.Error(err))
+		return err
+	}
+
+	// Get the list of filters on the link
+	filters, err := netlink.FilterList(link, netlink.HANDLE_MIN_INGRESS)
+	if err != nil {
+		logger.Error("Failed to get filter list", zap.Error(err))
+		return err
+	}
+
+	// Check if egress filter exists and delete it. Filter is identified by priority=1 and handle=1.
+	// this is to avoid duplicate filters after restarting the daemonset
+	for _, filter := range filters {
+		if filter.Attrs().Priority == 1 && filter.Attrs().Handle == 1 {
```

Comment:
what if kernel gives different handle id? can't we filter based on bpf prog name?